File: Backend/TranscomaAPI/Comun/Entidades/Shipper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TranscomaAPI.Comun.Entidades
{
    /// <summary>
    /// Entidad que registra los datos del Shipper
    /// del BL
    /// </summary>
    public class Shipper: Entidad
    {
        private string _nombre;
        private string _empresa;
        private string _nif;

        public Shipper(int id)
        {
            this.Id = id;
        }

        public Shipper(int id, string nombre, string empresa, string nif)
        {
            this.Id = id;
            this._nombre = nombre;
            this._empresa = empresa;
            this._nif = nif;
        }

        public string Nombre { get => _nombre; set => _nombre = value; }
        public string Empresa { get => _empresa; set => _empresa = value; }
        public string Nif { get => _nif; set => _nif = value; }
    }
}

File: frontend/src/app/models/shipper.ts
import { EntidadBase } from "./EntidadBase";

export class Shipper extends EntidadBase {

    private _nombre: string;
    private _empresa: string;
    private _nif: string;

    constructor(data?: any)  {
        super();
        if (data !== undefined) {
            this.$id = data._id;
            this.$empresa = data._empresa;
            this.$nif = data._nif;
        }
    }

    public get $nombre(): string {
        return this._nombre;
    }

    public set $nombre(value: string) {
        this._nombre = value;
    }

    public get $empresa(): string {
        return this._empresa;
    }

    public set $empresa(value: string) {
        this._empresa = value;
    }

    public get $nif(): string {
        return this._nif;
    }

    public set $nif(value: string) {
        this._nif = value;
    }
}